zipline/data/us_equity_minutes.py

```diff
@@ -11,27 +11,402 @@
+            from floats to an integer to fit within the np.uint32.
+
+            The default is 1000 to support pricing data which comes in to the
+            thousands place.
+
+        expectedlen : int
+            The expected length of the dataset, used when creating the initial
+            bcolz ctable.
+
+            If the expectedlen is not used, the chunksize and corresponding
+            compression ratios are not ideal.
+
+            Defaults to supporting 15 years of market data.
+
+            see: http://bcolz.blosc.org/opt-tips.html#informing-about-the-length-of-your-carrays
+        """
+        self._rootdir = rootdir
+        self._first_trading_day = first_trading_day
+        self._market_opens = market_opens[
+            market_opens.index.slice_indexer(start=self._first_trading_day)]
+        self._trading_days = market_opens.index
+        self._minutes_per_day = minutes_per_day
+        self._expectedlen = expectedlen
+        self._ohlc_ratio = ohlc_ratio
+
+        self._minute_index = _calc_minute_index(
+            self._market_opens, self._minutes_per_day)
+
+        self._write_metadata()
+
+    @property
+    def first_trading_day(self):
+        return self._first_trading_day
+
+    @property
+    def metadata_path(self):
+        return os.path.join(self._rootdir, METADATA_FILENAME)
+
+    def _write_metadata(self):
+        """
+        Write the metadata to a JSON file in the rootdir.
+
+        Values contained in the metadata are:
+        first_trading_day : string
+            'YYYY-MM-DD' formatted representation of the first trading day
+             available in the dataset.
+        minute_index : list of integers
+             nanosecond integer representation of the minutes, the enumeration of which
+             corresponds to the values in each bcolz carray.
+        """
+        metadata = {
+            'first_trading_day': str(self.first_trading_day.date()),
+            'minute_index': self._minute_index.asi8.tolist(),
+            'ohlc_ratio': self._ohlc_ratio,
+        }
+        with open(self.metadata_path, 'w+') as fp:
+            json.dump(metadata, fp)
+
+    def sidpath(self, sid):
+        """
+        Parameters:
+        -----------
+        sid : int
+            Asset identifier.
+
+        Returns:
+        --------
+        string
+
+            Full path to the bcolz rootdir for the given sid.
+        """
+        sid_subdir = _sid_subdir_path(sid)
+        return join(self._rootdir, sid_subdir)
+
+    def last_date_in_output_for_sid(self, sid):
+        """
+        Parameters:
+        -----------
+        sid : int
+            Asset identifier.
+
+        Returns:
+        --------
+        pd.Timestamp
+            The midnight of the last date written in to the output for the
+            given sid.
+        """
+        try:
+            sizes_path = "{0}/close/meta/sizes".format(self.sidpath(sid))
```

Comment:
This feels like it's relying a lot on bcolz implementation details.  I'd expect that a safer way to do this would be to make a `ctable` or `carray` and call `len` on it.  Does that not work and/or is it too slow?